core/templates/components/state-editor/state-editor-properties-services/state-editor.service.spec.ts

```diff
@@ -145,5 +231,323 @@ describe('Editor state service', () => {
+        customizationArgsForItemSelectionAndDragAndDropInput)
+    ).toEqual(null);
+  });
+
+  it('should return null when getting answer choices' +
+    ' if interactionID is empty', () => {
+    expect(ecs.getAnswerChoices('', {
+      choices: {
+        value: [
+          new SubtitledHtml('Choice 1', ''),
+          new SubtitledHtml('Choice 2', '')
+        ]
+      }
+    })).toBe(null);
+  });
+
+  it('should return if exploration is whitelisted or not', () => {
+    expect(ecs.isExplorationWhitelisted()).toBe(false);
+    ecs.explorationIsWhitelisted = true;
+    expect(ecs.isExplorationWhitelisted()).toBe(true);
+    ecs.explorationIsWhitelisted = false;
+    expect(ecs.isExplorationWhitelisted()).toBe(false);
+  });
+
+  it('should return if correctness feedback is enabled or not', () => {
+    expect(ecs.correctnessFeedbackEnabled).toBe(null);
+    ecs.correctnessFeedbackEnabled = true;
+    expect(ecs.correctnessFeedbackEnabled).toBe(true);
+    ecs.correctnessFeedbackEnabled = false;
+    expect(ecs.correctnessFeedbackEnabled).toBe(false);
+  });
+
+  it('should update state content editor initialised to true', () => {
+    expect(ecs.stateContentEditorInitialised).toBe(false);
+    ecs.updateStateContentEditorInitialised();
+    expect(ecs.stateContentEditorInitialised).toBe(true);
+  });
+
+  it('should update state interaction editor initialised to true', () => {
+    expect(ecs.stateInteractionEditorInitialised).toBe(false);
+    ecs.updateStateInteractionEditorInitialised();
+    expect(ecs.stateInteractionEditorInitialised).toBe(true);
+  });
+
+  it('should update state responses initialised to true', () => {
+    expect(ecs.stateResponsesInitialised).toBe(false);
+    ecs.updateStateResponsesInitialised();
+    expect(ecs.stateResponsesInitialised).toBe(true);
+  });
+
+  it('should update state hints editor initialised to true', () => {
+    expect(ecs.stateHintsEditorInitialised).toBe(false);
+    ecs.updateStateHintsEditorInitialised();
+    expect(ecs.stateHintsEditorInitialised).toBe(true);
+  });
+
+  it('should update state solution editor initialised to true', () => {
+    expect(ecs.stateSolutionEditorInitialised).toBe(false);
+    ecs.updateStateSolutionEditorInitialised();
+    expect(ecs.stateSolutionEditorInitialised).toBe(true);
+  });
+
+  it('should update state editor directive initialised to true', () => {
+    expect(ecs.stateEditorDirectiveInitialised).toBe(false);
+    ecs.updateStateEditorDirectiveInitialised();
+    expect(ecs.stateEditorDirectiveInitialised).toBe(true);
+  });
+
+  it('should update update current rule input is valid', () => {
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(false);
+    ecs.updateCurrentRuleInputIsValid(true);
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(true);
+    ecs.updateCurrentRuleInputIsValid(false);
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(false);
+  });
+
+  it('should set state names and emit state names change event', () => {
```

Comment:
Are we actually testing `..emit state names change event`?

Reply:
Changed test description